sequencefile/reader.go

```diff
@@ -0,0 +1,177 @@
+package sequencefile
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// A Reader reads key/value pairs from an input stream.
+type Reader struct {
+	Header          Header
+	syncMarkerBytes []byte
+
+	reader io.Reader
+	closed bool
+	err    error
+
+	buf   bytes.Buffer
+	key   []byte
+	value []byte
+}
+
+// New returns a new Reader for sequencefiles, reading input from r. If the
+// io.Reader is positioned at the start of a file, you should immediately call
+// ReadHeader to read through the header.
```

Comment:
Well, the nice thing is that you can use this to read input streams that don't have headers - or start in the middle of a file, like I do in the indexing code. It also means that `New` doesn't return an error, which I've been trying to get out of the habit of doing.